cli/src/state/atoms/keyboard.ts

```diff
@@ -310,10 +313,30 @@ export const submitInputAtom = atom(null, (get, set, text: string | Buffer) => {
 /**
  * Helper function to format autocomplete suggestions for display/submission
  */
-function formatSuggestion(suggestion: CommandSuggestion | ArgumentSuggestion, currentInput: string): string {
+function formatSuggestion(
+	suggestion: CommandSuggestion | ArgumentSuggestion | FileMentionSuggestion,
+	currentInput: string,
+	fileMentionContext?: { mentionStart: number; query: string } | null,
+): string {
 	if ("command" in suggestion) {
 		// CommandSuggestion - return full command with slash
 		return `/${suggestion.command.name}`
+	} else if ("type" in suggestion && (suggestion.type === "file" || suggestion.type === "folder")) {
+		// FileMentionSuggestion - insert file path at @ position with proper escaping
+		const fileSuggestion = suggestion as FileMentionSuggestion
+
+		if (!fileMentionContext) {
+			return currentInput
+		}
+
+		// Escape spaces in file path
+		const escapedPath = fileSuggestion.value.replace(/ /g, "\\ ")
```

Comment:
Spaces are valid in file paths, right? How does this handle a path like `~/Projects/kilocode/my folder/test.ts`?